### ISSUE TITLE
Lots of errors when running on Scale (but script seems to work)

### ISSUE THREAD

**Issue by rsteckler** [dak180/FreeNAS-Report#36]
Running on TrueNAS scale after manually installing bc.

The email works well, but I get the following when running in the shell:

```
sudo ./smart2.sh -c /home/admin/scripts/report.cfg
(standard_in) 1: syntax error
(standard_in) 2: illegal character: :
(standard_in) 2: syntax error
(standard_in) 3: illegal character: :
(standard_in) 4: illegal character: :
(standard_in) 5: illegal character: :
(standard_in) 6: illegal character: :
(standard_in) 7: illegal character: :
(standard_in) 8: illegal character: :
(standard_in) 9: illegal character: :
(standard_in) 9: syntax error
(standard_in) 10: illegal character: :
(standard_in) 11: illegal character: :
(standard_in) 12: illegal character: :
(standard_in) 13: illegal character: :
(standard_in) 14: illegal character: :
(standard_in) 15: illegal character: :
(standard_in) 16: illegal character: :
(standard_in) 17: syntax error
(standard_in) 18: illegal character: :
(standard_in) 19: illegal character: :
(standard_in) 19: syntax error
(standard_in) 20: illegal character: :
(standard_in) 22: syntax error
(standard_in) 1: syntax error
(standard_in) 1: syntax error
(standard_in) 1: syntax error
(standard_in) 2: illegal character: :
(standard_in) 2: syntax error
(standard_in) 3: illegal character: :
(standard_in) 4: illegal character: :
(standard_in) 5: illegal character: :
(standard_in) 6: illegal character: :
(standard_in) 7: illegal character: :
(standard_in) 8: illegal character: :
(standard_in) 9: illegal character: :
(standard_in) 9: syntax error
(standard_in) 10: illegal character: :
(standard_in) 11: illegal character: :
(standard_in) 12: illegal character: :
(standard_in) 13: illegal character: :
(standard_in) 14: illegal character: :
(standard_in) 15: illegal character: :
(standard_in) 16: illegal character: :
(standard_in) 17: syntax error
(standard_in) 18: illegal character: :
(standard_in) 19: illegal character: :
(standard_in) 19: syntax error
(standard_in) 20: illegal character: :
(standard_in) 22: syntax error
(standard_in) 22: syntax error
```

**Comment by dak180** [comment]
How did you download the file?

**Comment by rsteckler** [comment]
View raw on GitHub, copy, then pasted into vi in a shell on the truenas box.

A quick search shows that these errors are from bc and not from bash.

I can try wget, but just left town for a few days and won't be back until Thursday

**Comment by dak180** [comment]
When you can test that, that would be great; I would like to eliminate the possibility of greebles being introduced.

**Comment by rsteckler** [comment]
Same issue when cloning from github:
```
#------------------------------------ Show the report.cfg location
admin@truenas[~/scripts]$ pwd
/home/admin/scripts

#------------------------------------ Show the report.cfg file
admin@truenas[~/scripts]$ cat report.cfg

# Set this to 0 to enable
defaultFile="0"

###### User-definable Parameters
### Email Address
email="myemail@example.org" #---- redacted

### Global table colors
okColor="#c9ffcc"       # Hex code for color to use in SMART Status column if drives pass (default is light green, #c9ffcc)
warnColor="#ffd6d6"     # Hex code for WARN color (default is light red, #ffd6d6)
critColor="#ff0000"     # Hex code for CRITICAL color (default is bright red, #ff0000)
altColor="#f4f4f4"      # Table background alternates row colors between white and this color (default is light gray, #f4f4f4)

### zpool status summary table settings
usedWarn="90"                   # Pool used percentage for CRITICAL color to be used
scrubAgeWarn="30"               # Maximum age (in days) of last pool scrub before CRITICAL color will be used

### SMART status summary table settings
includeSSD="true"               # Change to "true" to include SSDs in SMART status summary table; "false" to disable
includeSAS="false"              # Change to "true" to include SAS drives in SMART status summary table; "false" to disable
lifeRemainWarn="75"             # Life remaining in the SSD at which WARNING color will be used
lifeRemainCrit="50"             # Life remaining in the SSD at which CRITICAL color will be used
totalBWWarn="100"               # Total bytes written (in TB) to the SSD at which WARNING color will be used
totalBWCrit="200"               # Total bytes written (in TB) to the SSD at which CRITICAL color will be used
tempWarn="35"                   # Drive temp (in C) at which WARNING color will be used
tempCrit="40"                   # Drive temp (in C) at which CRITICAL color will be used
ssdTempWarn="40"                # SSD drive temp (in C) at which WARNING color will be used
ssdTempCrit="45"                # SSD drive temp (in C) at which CRITICAL color will be used
sectorsCrit="10"                # Number of sectors per drive with errors before CRITICAL color will be used
testAgeWarn="5"                 # Maximum age (in days) of last SMART test before CRITICAL color will be used
powerTimeFormat="ymdh"  # Format for power-on hours string, valid options are "ymdh", "ymd", "ym", or "y" (year month day hour)

### TrueNAS config backup settings
configBackup="false"                    # Change to "false" to skip config backup (which renders next two options meaningless); "true" to keep config backups enabled
emailBackup="false"                             # Change to "true" to email TrueNAS config backup
saveBackup="true"                               # Change to "false" to delete TrueNAS config backup after mail is sent; "true" to keep it in dir below
backupLocation="/root/backup"   # Directory in which to save TrueNAS config backups

### UPS status summary settings
reportUPS="false"                       # Change to "false" to skip reporting the status of the UPS

### General script settings
logfileLocation="/tmp"          # Directory in which to save TrueNAS log file. Can be set to /tmp.
logfileName="logfilename"       # Log file name
saveLogfile="true"                      # Change to "false" to delete the log file after creation

##### Drive Overrides
# In the form: declare -A _<serial>
# And then for each override: _<serial>[<value>]="<adjustment>"
# Replace any non ascii alphanumeric characters with _ in the serial.

#------------------------------------ Clone the repo
admin@truenas[~/scripts]$ git clone https://github.com/dak180/FreeNAS-Report.git
Cloning into 'FreeNAS-Report'...
remote: Enumerating objects: 772, done.
remote: Counting objects: 100% (210/210), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 772 (delta 180), reused 175 (delta 152), pack-reused 562 (from 1)
Receiving objects: 100% (772/772), 415.45 KiB | 2.56 MiB/s, done.
Resolving deltas: 100% (475/475), done.

#------------------------------------ Change into the repo directory
admin@truenas[~/scripts]$ cd FreeNAS-Report 

#------------------------------------ Run the script
admin@truenas[~/scripts/FreeNAS-Report]$ sudo ./report.sh -c /home/admin/scripts/report.cfg    
(standard_in) 1: syntax error
(standard_in) 2: illegal character: :
(standard_in) 2: syntax error
(standard_in) 3: illegal character: :
(standard_in) 4: illegal character: :
(standard_in) 5: illegal character: :
(standard_in) 6: illegal character: :
(standard_in) 7: illegal character: :
(standard_in) 8: illegal character: :
(standard_in) 9: illegal character: :
(standard_in) 9: syntax error
(standard_in) 10: illegal character: :
(standard_in) 11: illegal character: :
(standard_in) 12: illegal character: :
(standard_in) 13: illegal character: :
(standard_in) 14: illegal character: :
(standard_in) 15: illegal character: :
(standard_in) 16: illegal character: :
(standard_in) 17: syntax error
(standard_in) 18: illegal character: :
(standard_in) 19: illegal character: :
(standard_in) 19: syntax error
(standard_in) 20: illegal character: :
(standard_in) 22: syntax error
(standard_in) 1: syntax error
(standard_in) 1: syntax error
(standard_in) 1: syntax error
(standard_in) 2: illegal character: :
(standard_in) 2: syntax error
(standard_in) 3: illegal character: :
(standard_in) 4: illegal character: :
(standard_in) 5: illegal character: :
(standard_in) 6: illegal character: :
(standard_in) 7: illegal character: :
(standard_in) 8: illegal character: :
(standard_in) 9: illegal character: :
(standard_in) 9: syntax error
(standard_in) 10: illegal character: :
(standard_in) 11: illegal character: :
(standard_in) 12: illegal character: :
(standard_in) 13: illegal character: :
(standard_in) 14: illegal character: :
(standard_in) 15: illegal character: :
(standard_in) 16: illegal character: :
(standard_in) 17: syntax error
(standard_in) 18: illegal character: :
(standard_in) 19: illegal character: :
(standard_in) 19: syntax error
(standard_in) 20: illegal character: :
(standard_in) 22: syntax error
(standard_in) 22: syntax error

admin@truenas[~/scripts/FreeNAS-Report]$ 
```

Note that the script seems to work great, so this isn't getting in my way at all.  Just wanted you to be aware in case it indicates something more troublesome behind the scenes.  

The email report looks good to me, but happy to forward to you so you can look at it if that's helpful.

**Comment by dak180** [comment]
Can you attach a dump file `sudo ./smart2.sh -dc /home/admin/scripts/report.cfg`?

**Comment by rsteckler** [comment]
Here ya go (note github doesn't allow .tgz for some reason, so recompressed as .zip)

[dumpfiles.zip](https://github.com/user-attachments/files/16938203/dumpfiles.zip)

**Comment by dak180** [comment]
Try 2d274f630eb92e2a4ab833768630da024e48ab8e and tell me how that works for you.

**Comment by rsteckler** [comment]
That worked!  Zero errors now.

**Comment by dak180** [comment]
Fixed in 2d274f630eb92e2a4ab833768630da024e48ab8e.
